src/lib/pages/validator-details/components/tables/ProposedBlocksTable.tsx

```diff
@@ -55,72 +136,66 @@ export const ProposedBlocksTable = ({
   );
 
   if (isLoading) return <Loading />;
-  if (error) return <ErrorFetching dataName="blocks" />;
-  if (!data?.total)
+  if (error || !data) return <ErrorFetching dataName="blocks" />;
+  if (!data.total)
     return (
-      <EmptyState
-        imageVariant="empty"
-        message="This validator never proposed any blocks."
-        withBorder
-      />
+      <>
+        <TableTitle
+          title="Proposed Blocks"
+          count={0}
+          helperText={
+            onViewMore
+              ? ""
+              : "Display the proposed blocks by this validator within the last 30 days"
+          }
+          mb={0}
+        />
+        <EmptyState
+          imageVariant={onViewMore ? undefined : "empty"}
+          message="This validator never proposed any blocks."
+          withBorder
+        />
+      </>
     );
 
   return (
     <>
-      <TableTitle
-        title="Proposed Blocks"
-        count={data?.total}
-        mt={2}
-        mb={1}
-        helperText="Display the proposed blocks by this validator within the last 30 days"
-      />
-      {isMobile ? (
-        <MobileTableContainer>
-          {data.items.map((block) => (
-            <BlocksTableMobileCard
-              key={block.hash}
-              blockData={block}
-              hideProposer
-            />
-          ))}
-        </MobileTableContainer>
-      ) : (
-        <TableContainer>
-          <BlocksTableHeader
-            templateColumns={TEMPLATE_COLUMNS}
-            scrollComponentId={scrollComponentId}
-            hideProposer
-          />
-          {data.items.map((block) => (
-            <BlocksTableRow
-              key={block.hash}
-              templateColumns={TEMPLATE_COLUMNS}
-              blockData={block}
-              hideProposer
-            />
-          ))}
-        </TableContainer>
-      )}
-      {onViewMore && data.total > 5 && (
-        <ViewMore
+      {isMoibleOverview ? (
+        <Flex
+          backgroundColor="gray.900"
+          p={4}
+          rounded={8}
+          w="100%"
+          justifyContent="space-between"
+          alignItems="center"
           onClick={onViewMore}
-          text={`View all proposed blocks (${data.total})`}
+        >
+          <TableTitle title="Proposed Blocks" count={data.total} mb={0} />
+          <CustomIcon boxSize={6} m={0} name="chevron-right" color="gray.600" />
+        </Flex>
+      ) : (
+        <TableTitle
+          title="Proposed Blocks"
+          count={data.total}
+          helperText={
+            onViewMore
+              ? ""
+              : "Display the proposed blocks by this validator within the last 30 days"
+          }
+          mb={0}
         />
       )}
-      {!onViewMore && data.total > 10 && (
-        <Pagination
+      {(!isMobile || !onViewMore) && (
```

Comment:
- remove this condition and put this right after `TableTitle`
- let's keep `viewmore` and `pagination` in this component